pisa/analysis/analysis.py

```diff
@@ -2456,6 +2477,7 @@ def _minimizer_callable(self, scaled_param_vals, hypo_maker, data_dist,
         #
         try:
             if hypo_maker.__class__.__name__ == "Detectors":
+                hypo_maker.update_params(hypo_maker.params)
```

Comment:
I don't understand this line... why update the parameters with the parameters that it already has? Be sure that this is what you want: The `update_params` method replaces the references inside the `hypo_maker`, which can lead to unintended consequences! If you intend to change the *values*, please use the `update_param_values` function defined in this module.

Reply:
@JanWeldert Could you please come back to this? It is my last question, once we have sorted this line out we are good to go.

Reply:
Ok, I see now. I have never worked with multiple detectors before, but at least from a cursory reading it checks out.